### PR TITLE
Move `mypy` + `isort` + `flake8` config from `.pre-commit-config.yaml` to `setup.cfg`

### PR DESCRIPTION
**Pull request by janosh**
Same treatment as in https://github.com/materialsproject/atomate2/commit/6270579a615aa333ad0b8348ce8fcd55dadce59a. Plus some typo fixes.

### CODE REVIEW

File: src/jobflow/managers/local.py [hunk 1]
```diff
@@ -165,4 +165,4 @@ def _run(root_flow):
     if ensure_success and not finished_successfully:
         raise RuntimeError("Flow did not finish running successfully")
 
-    return dict(responses)
+    return responses
```
Comment (utf) [inline src/jobflow/managers/local.py:168]: This is actually purposeful to convert from defaultdict to dict. Purely a style thing but I think this should stay as is?

**Review by utf — COMMENTED** [review]
Thanks! Just one minor comment.

File: src/jobflow/managers/local.py (hunk 1 above)
Comment (janosh) [reply src/jobflow/managers/local.py:168]: `defaultdict` subclasses `dict` so for the purpose of `mypy`, there's no need to convert. At run time `defaultdict` offers all the methods of `dict` too so no potential for problems there either (afaik). But happy to revert if you prefer.

Comment (utf) [reply src/jobflow/managers/local.py]: Yeah, its not for the API, functionality, or mypy. Just purely stylistic when working with jobflow in a jupyter notebook for example. Thanks for reverting.

**Comment by utf** [comment]
Thanks!

**Pull request merged**